### PR TITLE
Feat: add apply component definition for docs

### PR DESCRIPTION
**Pull request by FogDong**
Signed-off-by: FogDong <dongtianxin.tx@alibaba-inc.com>

### Description of your changes

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: charts/vela-core/templates/defwithtemplate/apply-component.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    custom.definition.oam.dev/ui-hidden: "true"
+  name: apply-component
+  namespace: {{ include "systemDefinitionNamespace" . }}
+spec:
+  schematic:
+    cue:
+      template: |
+        parameter: {
+        	// +usage=Specify the component name to apply
+        	component: string
```
Comment (wonderflow) [inline charts/vela-core/templates/defwithtemplate/apply-component.yaml:19]: apply component also support specify cluster?

File: vela-templates/definitions/internal/workflowstep/apply-component.cue [hunk 2]
```diff
@@ -0,0 +1,15 @@
+"apply-component": {
+	type: "workflow-step"
+	annotations: {}
+	labels: {
+		"ui-hidden": "true"
+		"scope":     "Application"
+	}
+	description: "Apply a component and its corresbonding traits in application"
```
Comment (wonderflow) [inline vela-templates/definitions/internal/workflowstep/apply-component.cue:8]:
```suggestion
	description: "Apply a specific component and its corresbonding traits in application"
```

File: charts/vela-core/templates/defwithtemplate/apply-component.yaml (hunk 1 above)
Comment (FogDong) [reply charts/vela-core/templates/defwithtemplate/apply-component.yaml:19]: No, `cluster` is valid for `op.#ApplyComponent` but not `apply-component` def.

File: references/docgen/def-doc/workflowstep/apply-component.eg.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        domain: testsvc.example.com
+        http:
+          /: 8000
+  workflow:
+    steps:
+      - name: express-server
+        type: apply-component
+        properties:
+          component: express-server
+```
```
Comment (wonderflow) [inline references/docgen/def-doc/workflowstep/apply-component.eg.md:26]: let's add another example to specify the cluster?

**Review by wonderflow — APPROVED** [review]

**Review by Somefive — APPROVED** [review]

**Pull request merged**